config/reader.go

```diff
@@ -610,7 +610,24 @@ func processConfig(unprocessedConfig *Config, fromCloud bool, logger logging.Log
 		}
 	}
 
-	if err := logging.UpdateLoggerRegistry(cfg.LogConfig); err != nil {
+	appendedLogCfg := make([]logging.LoggerPatternConfig, 0, len(cfg.LogConfig)+len(cfg.Services)+len(cfg.Components))
+	appendedLogCfg = append(appendedLogCfg, cfg.LogConfig...)
+	for _, serv := range cfg.Services {
```

Comment:
I'm assuming this `append` during `processConfig` is to maintain priority of `log_configuration` over the other user-specified patterns? Can we leave a comment above this hard-coding explaining that?

Reply:
it's the opposite, resource configurations take priority over pattern matching configurations, so we append those patterns to the end of the logger level configs. This way, when we process in top down order, the semantics we defined earlier hold.